### PR TITLE
expression: empty CURRENT_ROLE

### LINKED ISSUE #15199: different behavior of `select CURRENT_ROLE()` between TiDB and MySQL 8.0
## General Question
 
TiDB

```
mysql> SELECT CURRENT_ROLE();
+----------------+
| CURRENT_ROLE() |
+----------------+
|                |
+----------------+
```

MySQL 8.0

```
mysql> SELECT CURRENT_ROLE();
+----------------+
| CURRENT_ROLE() |
+----------------+
| NONE           |
+----------------+
```

### PR DESCRIPTION
**Pull request by tisonkun**
**co https://github.com/pingcap/tidb-test/pull/1219**

Signed-off-by: tison <wander4096@gmail.com>

### What problem does this PR solve?

Issue Number: close #15199

Problem Summary:

### What is changed and how it works?

What's Changed:

Cover the case where `CURRENT_ROLE` is none.

How it Works:

Straightforward.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests

- Integration test

### Release note 

- `CURRENT_ROLE` now returns `NONE` with empty roles setting.

### CODE REVIEW

File: expression/builtin_info.go [hunk 1]
```diff
@@ -217,17 +217,16 @@ func (b *builtinCurrentRoleSig) Clone() builtinFunc {
 	return newSig
 }
 
-// evalString evals a builtinCurrentUserSig.
-// See https://dev.mysql.com/doc/refman/5.7/en/information-functions.html#function_current-user
-func (b *builtinCurrentRoleSig) evalString(row chunk.Row) (string, bool, error) {
+// evalString eval a builtinCurrentUserSig.
```
Comment (SunRunAway) [inline expression/builtin_info.go:220]: evals?

File: expression/builtin_info.go [hunk 2]
```diff
@@ -217,17 +217,16 @@ func (b *builtinCurrentRoleSig) Clone() builtinFunc {
...
+// See https://dev.mysql.com/doc/refman/8.0/en/information-functions.html#function_current-role
+func (b *builtinCurrentRoleSig) evalString(row chunk.Row) (res string, isNil bool, err error) {
```
Comment (SunRunAway) [inline expression/builtin_info.go:222]:
```suggestion
func (b *builtinCurrentRoleSig) evalString(row chunk.Row) (res string, isNull bool, err error) {
```

File: server/server_test.go [hunk 3]
```diff
@@ -1684,6 +1684,35 @@ func (cli *testServerClient) runTestIssue3682(c *C) {
 	c.Assert(err.Error(), Equals, "Error 1045: Access denied for user 'issue3682'@'127.0.0.1' (using password: YES)")
 }
 
+func (cli *testServerClient) runTestEmptyRoles(c *C) {
```
Comment (SunRunAway) [inline server/server_test.go:1687]: It's better to put the test into the same package, like expression/integration_test.go or expression/buildin_info_test.go

File: expression/builtin_info_vec.go [hunk 4]
```diff
@@ -122,8 +122,8 @@ func (b *builtinCurrentRoleSig) vectorized() bool {
 	return true
 }
 
-// evalString evals a builtinCurrentUserSig.
-// See https://dev.mysql.com/doc/refman/5.7/en/information-functions.html#function_current-user
+// evalString eval a builtinCurrentUserSig.
```
Comment (SunRunAway) [inline expression/builtin_info_vec.go:125]: evals

File: expression/builtin_info.go (hunk 1 above)
Comment (tisonkun) [reply expression/builtin_info.go:220]:
```suggestion
// evalString evals a builtinCurrentUserSig.
```

File: server/server_test.go (hunk 3 above)
Comment (tisonkun) [reply server/server_test.go:1687]: ok. Let me have a try. I just add a test where `SELECT CURRENT_ROLE` occurs and it is here.

File: expression/builtin_info_vec.go (hunk 4 above)
Comment (tisonkun) [reply expression/builtin_info_vec.go:125]:
```suggestion
// evalString evals a builtinCurrentUserSig.
```

**Comment by tisonkun** [comment]
<br class="Apple-interchange-newline">/run-all-tests tidb-test=pr/1219

File: server/server_test.go (hunk 3 above)
Comment (tisonkun) [reply server/server_test.go:1687]: `expression/integration_test.go` or `expression/buildin_info_test.go` doesn't test `CURRENT_ROLE` raw output, and here is the original test. I'd prefer adding tests as is and delay the test reorder later.

Comment (tisonkun) [reply server/server_test.go:1687]: I found that we do unit tests in these files. Will add more tests.

Comment (tisonkun) [reply server/server_test.go]: done 636205971801ad8d3f0551c307c3626835d3419d

**Review by wshwsh12 — APPROVED** [review]

**Review by wshwsh12 — APPROVED** [review]

**Review by SunRunAway — APPROVED** [review]

**Comment by tisonkun** [comment]
@SunRunAway @wshwsh12 please take a look at the corresponding pr on tidb-test pingcap/tidb-test#1219

**Pull request merged**